pgbouncer_custom/pgbouncer_custom.py

```diff
@@ -0,0 +1,148 @@
+                    self.log.exception("Not all metrics may be available")
+                else:
+                    count_by_db_and_host = defaultdict(lambda: defaultdict(int))
+                    for row in rows:
+                        self.log.debug("Processing row: %r", row)
+                        db_name = row['database']
+                        client_addr = row['addr']
+                        count_by_db_and_host[client_addr][db_name] += 1
+                    for client, counts_by_host in count_by_db_and_host.iteritems():
+                        for db_name, count in counts_by_host.iteritems():
+                            tags = {'client_addr': client, 'db_name': db_name}
+                            tags.extend(instance_tags)
+                            self.gauge(
+                                'pgbouncer_custom.clients.count',
+                                count,
+                                tags=tags
+                            )
+                    if not rows:
+                        self.log.warning("No results were found for query: %s", self.QUERY)
+        except pg.Error:
+            self.log.exception("Connection error")
+            raise ShouldRestartException
+
+    def _get_connect_kwargs(self, host, port, user, password, database_url):
+        """
+        Get the params to pass to psycopg2.connect() based on passed-in vals
+        from yaml settings file
+        """
+        if database_url:
+            return {'dsn': database_url}
+
+        if host in ('localhost', '127.0.0.1') and password == '':
+            return {  # Use ident method
+                'dsn': "user={} dbname={}".format(user, self.DB_NAME)
+            }
+
+        if port:
+            return {'host': host, 'user': user, 'password': password,
+                    'database': self.DB_NAME, 'port': port}
+
+        return {'host': host, 'user': user, 'password': password,
+                'database': self.DB_NAME}
+
+    def _get_connection(self, key, host='', port='', user='',
+                        password='', database_url='', tags=None, use_cached=True):
+        if key in self.dbs and use_cached:
+            return self.dbs[key]
+        try:
+            connect_kwargs = self._get_connect_kwargs(
+                host=host, port=port, user=user,
+                password=password, database_url=database_url
+            )
+            connection = pg.connect(**connect_kwargs)
+            connection.set_isolation_level(
+                pg.extensions.ISOLATION_LEVEL_AUTOCOMMIT)
+        except Exception:
+            redacted_url = self._get_redacted_dsn(host, port, user, database_url)
+            message = u'Cannot establish connection to {}'.format(redacted_url)
+            self.service_check(self.SERVICE_CHECK_NAME, AgentCheck.CRITICAL,
+                               tags=self._get_service_checks_tags(host, port, database_url, tags),
+                               message=message)
```

Comment:
I don't think we need already have a pgbouncer service check

Reply:
Sure